### ISSUE TITLE
Start exporting exe and deb?

### ISSUE THREAD

**Issue by baseplate-admin** [hitblast/avnie#2]
Hi i think, exe and deb and dmg should be the next logical step

**Comment by hitblast** [comment]
@baseplate-admin Yeah, me too. Do you have any ideas on how to get the binaries started? I'm also gonna attach CI/CD with it.

**Comment by hitblast** [comment]
@baseplate-admin When you're free, have a look at the changes here: https://github.com/hitblast/avnie/pull/3/files
I'm currently working on creating the individual build options and the `avnie/pyinstaller.py` script for automating the build process.

**Comment by hitblast** [comment]
It's worth mentioning that there's a bit of work needed to be completed first regarding the binary creation since pyinstaller is throwing an error regarding a non-existent version file.

**Comment by baseplate-admin** [comment]
> @baseplate-admin When you're free, have a look at the changes here: https://github.com/hitblast/avnie/pull/3/files

my apologies, ami ektu research niye besto chilam.


>  I'm currently working on creating the individual build options and the avnie/pyinstaller.py script for automating the build process.

You should look into github actions. and then run file for each of the os.

```python
pyinstaller --onefile src/main.py
```

Run this for all of the OS and then upload the action using [upload-artifact](https://github.com/actions/upload-artifact).

**Comment by hitblast** [comment]
@baseplate-admin Was alll fun and games until I discovered that pyinstaller initializes an unwanted hook while building the executable, which doesn't resemble avro.py but has the exact same import name. So, it fails to identify between both libraries. If a successful build is to be created with pyinstaller, we'll have to modify the hook associated with pyinstaller entirely.

This is kind of a very hard option.

And even after modifying the hook (I've tried inside a virtual environment manually), the built executable takes 15-20 seconds to even start running. So, this brings back that old question of switching backends to a C-compliant language.
